### ISSUE TITLE
MW-CSRF ignores POST/PUT/DELETE with Content-Type: application/json

### ISSUE THREAD

**Issue by MaMrEzO** [gobuffalo/mw-csrf#6]
I'm developing backend via GoBuffalo, hence I need CSRF check for `POST/DELETE/PUT` methods but MW-CSRF simply ignore these requests with `Content-Type: application/json` due to:

```golang
  htmlTypes   = []string{"html", "form", "plain", "*/*"}
  .
  .
  .
  req := c.Request()
  ct := req.Header.Get("Content-Type")
  if len(ct) == 0 {
    ct = req.Header.Get("Accept")
  }
  // ignore non-html requests
  if ct != "" && !contains(htmlTypes, ct) {
    return next(c)
  }
```
Note: I'm using `"github.com/rs/cors"` too, the FrontEnd is under development using ReactJS.
What could I do?

**Comment by moshid** [comment]
Hi @MaMrEzO , what did you end up doing?

**Comment by Bricklou** [comment]
This issue is a bit dead but yes, `application/json` is not supported and cause some issues for custom API.
The module really need some configurable options :confused:

**Comment by sio4** [comment]
Indeed, not just `application/json`, I can't imagine the reason for checking `Content-Type` for CSRF protection. I checked with the history and that code was introduced by PR gobuffalo/buffalo#387 without description or linked issue. After that, it was extended to support `*/*` by the issue gobuffalo/buffalo#767 which was fixed by gobuffalo/buffalo#767 but... the history is not enough.

Anyway, as far as I know, CSRF is not relevant to the Content-Type. If there is any chance that a manipulated request can make a change to the data (modify or delete or anything) with an HTTP request, there is also a chance of a CSRF attack.

Hi @paganotoni, Do you know if there is any history/reason for it? Otherwise, I will remove the content-type checking and will check some other issues.
